client/app/api/auth/[...nextauth]/route.ts

```diff
@@ -0,0 +1,144 @@
+                Body type: x-www-form-urlencoded
+                Form fields:
+                client_id :
+                client_secret :
+                grant_type : refresh_token
+                refresh_token:
+              */
+              const details = {
+                client_id: `${process.env.KEYCLOAK_CLIENT_ID}`,
+                client_secret: `${process.env.KEYCLOAK_CLIENT_SECRET}`,
+                grant_type: ["refresh_token"],
+                refresh_token: token.refresh_token,
+              };
+              const formBody: string[] = [];
+              Object.entries(details).forEach(([key, value]: [string, any]) => {
+                const encodedKey = encodeURIComponent(key);
+                const encodedValue = encodeURIComponent(value);
+                formBody.push(encodedKey + "=" + encodedValue);
+              });
+              const formData = formBody.join("&");
+              const url = `${process.env.KEYCLOAK_TOKEN_URL}`;
+              const response = await fetch(url, {
+                method: "POST",
+                headers: {
+                  "Content-Type":
+                    "application/x-www-form-urlencoded;charset=UTF-8",
+                },
+                body: formData,
+              });
+              const refreshedToken = await response.json();
+              if (!response.ok) throw refreshedToken;
+              return {
+                ...token,
+                error: refreshedToken.error,
+                access_token: refreshedToken.access_token,
+                access_token_expires_at:
+                  Math.floor(Date.now() / 1000) +
+                  (refreshedToken.refresh_expires_in ?? 0),
+                refresh_token:
+                  refreshedToken.refresh_token ?? token.refresh_token, // Fall back to old refresh token
+              };
+            } catch (error) {
+              token.error = "ErrorAccessToken";
+            }
+          }
+        }
+      } catch (error) {
+        token.error = "ErrorAccessToken";
+      }
+      // 🔒 return encrypted nextauth JWT
+      return token;
+    },
+    /**
+     * @param  {object} session      Session object
+     * @param  {object} token        User object    (if using database sessions)
+     *                               JSON Web Token (if not using database sessions)
+     * @return {object}              Session that will be returned to the client
+     */
+    // 👇️ called whenever a session is checked
+    async session({ session, token }) {
+      //🚨 🚨  By default, for security, only a subset of the token is returned...
+      //💡 if you want to make a nextauth JWT property available to the client session...
+      // you have to explicitly forward it here to make it available to the client
+      //🚨 🚨  Do not expose sensitive information, such as access tokens.
+      return {
+        ...session,
+        error: token.error,
+        user: {
+          ...session.user,
+          role: token.role,
```

Comment:
I was trying to add the guid to this object, but wasn't able to retrieve it in a component (with a console.log of session). Are you able to retrieve the guid from the session?

Reply:
See latest code push: ...idir_user_guid available to the session.
Let me know if you want to discuss.